### ISSUE TITLE
Support for auto-scanning files

### ISSUE THREAD

**Issue by ixrock** [willofindie/vscode-cssvar#12]
Would be nice to have an auto scanning all style files for variables in the project.

_Reason:_ we have a lot of components with custom css properties in separate files.
Therefore maintaining the list manually in the settings doesn't work well in that case..

_Solution:_ `chokidar` + `glob` (to limit directories for performance if needed). Wdyt?

**Comment by ixrock** [comment]
Tried this in the settings, doesn't work or works partially..

```json
    "cssvar.files": [
        "src/renderer/**/*.css",
        "src/renderer/**/*.scss",
    ],

```

**Comment by phoenisx** [comment]
This plugin, as of now, works only with pure css files. SCSS files are supported only if there's no sass specific code present in the file.

Anyways, good idea, will try to find solution for this

**Comment by ronilaukkarinen** [comment]
Got excited but then sad when noticed it doesn't work for SCSS. :( Also our files vary for each project so it's a pain defining them all the time. When I define it like this:

```
"cssvar.files": ["content/themes/**/*.css"],
```

I get:

```
Failed to parse CSS variables in files`

**Comment by phoenisx** [comment]
Are you folks using some custom postcss plugins??

If so, this plugin is kept minimum, as of now, and uses postcss parser, without any plugins, thus any rule not part of W3 CSS specification will through a parser error.

I am plaaning to support a config to provide postcss plugin in future, but faced some issues with implementing it, so have deferred the task for now.

Will update here, once I add the support for postcss plugins

**Comment by ronilaukkarinen** [comment]
Yes, using stylelint + postcss + etc. But it's not exactly clear to me why this plugin should support those as long as it just trieds to fetch CSS vars from the files.

**Comment by phoenisx** [comment]
To fetch variables from CSS files, the plugin needs to parse the files passed in `cssvar.files`, where if the files are not pure CSS, the parser will throw an error.
That's the reason I mentioned before this plugin doesn't support Sass or any custom variant of CSS, to keep the plugin small and less complex.

_But, I will in future add support to provide [Postcss Plugins](https://www.postcss.parts/)_

I hope this clears your doubt @ronilaukkarinen

**Comment by ronilaukkarinen** [comment]
Ah, OK, thanks for the clarification. Those CSS files I defined are pure CSS files understandable by browsers though.

**Comment by phoenisx** [comment]
Can you provide me with a sample file if possible? This will help me to debug the issue, if any!

**Comment by phoenisx** [comment]
I have released `v1.1` which adds support for CSS extensions like `SCSS` syntax and more.

Latest release now supports the following:
- `@use "file"` syntax
- `@import "file"` syntax
- Some custom SCSS/SASS syntax in CSS, use Postcss Plugin as mentioned here: [Plugin Customization](https://github.com/willofindie/vscode-cssvar/blob/v1.1.0/customize-extension.md#enhanced-css-using-postcss-plugins)
- If your project is entirely built with `SCSS` or some other CSS extension, use PostCSS syntax as mentioned here: [Syntax Customization](https://github.com/willofindie/vscode-cssvar/blob/v1.1.0/customize-extension.md#adding-support-for-any-new-sytax)
